src/atmos_param/socrates/interface/socrates_interface.F90

```diff
@@ -0,0 +1,929 @@
+  INTEGER :: n_soc_bands_lw, n_soc_bands_sw
+  INTEGER :: n_soc_bands_lw_hires, n_soc_bands_sw_hires
+  CHARACTER(len=10), PARAMETER :: soc_mod_name = 'socrates'
+  REAL :: missing_value = -999
+
+  type(interpolate_type),save                :: o3_interp, co2_interp            ! use external file for ozone and co2
+
+  REAL :: dt_last !Time of last radiation calculation - used to tell whether it is time to recompute radiation or not
+  REAL(r_def), allocatable, dimension(:,:,:) :: tdt_soc_sw_store, tdt_soc_lw_store, thd_sw_flux_down_store, thd_lw_flux_up_store
+  REAL(r_def), allocatable, dimension(:,:) :: net_surf_sw_down_store, surf_lw_down_store, coszen_store
+  REAL(r_def), allocatable, dimension(:,:,:) :: outputted_soc_spectral_olr
+
+CONTAINS
+
+  SUBROUTINE socrates_init(is, ie, js, je, num_levels, axes, Time, lat, lonb, latb, delta_t_atmos)
+    !! Initialises Socrates spectra, arrays, and constants
+
+    USE astronomy_mod, only: astronomy_init
+    USE interpolator_mod, only: interpolate_type, interpolator_init, ZERO  
+    USE socrates_config_mod
+
+    ! Arguments
+    INTEGER, INTENT(in), DIMENSION(4) :: axes
+    !! NB axes refers to the handles of the axes defined in fv_diagnostics
+    TYPE(time_type), INTENT(in)       :: Time, delta_t_atmos
+    INTEGER, INTENT(in)               :: is, ie, js, je, num_levels
+    REAL, INTENT(in) , DIMENSION(:,:)   :: lat
+    REAL, INTENT(in) , DIMENSION(:,:)   :: lonb, latb
+        
+    integer :: io, stdlog_unit
+    integer :: res, time_step_seconds
+    real    :: day_in_s_check
+
+    !-------------------------------------------------------------------------------------
+
+#ifdef INTERNAL_FILE_NML
+   read (input_nml_file, nml=socrates_rad_nml, iostat=io)
+#else
+   if ( file_exist('input.nml') ) then
+      nml_unit = open_namelist_file()
+      read (nml_unit, socrates_rad_nml, iostat=io)
+      call close_file(nml_unit)
+   endif
+#endif
+stdlog_unit = stdlog()
+write(stdlog_unit, socrates_rad_nml)
+
+    !Initialise astronomy
+    call astronomy_init
+
+    !Initialise variables related to radiation timestep
+
+          call get_time(delta_t_atmos,time_step_seconds)
+
+          if (dt_rad .le. 0.) then
+              dt_rad = time_step_seconds !Make sure that dt_rad is set if it is not specified in the namelist
+          endif
+
+          dt_last = -real(dt_rad) !make sure we are computing radiation at the first time step
+
+            if (dt_rad .gt. time_step_seconds) then
+                res=mod(dt_rad, time_step_seconds)
+
+                if (res.ne.0) then
+                        call error_mesg( 'socrates_init', &
+                         'dt_rad must be an integer multiple of dt_atmos',FATAL)
+                endif
+
+                day_in_s_check=length_of_day()
+                res=mod(int(day_in_s_check), dt_rad)
+
+                if (res.ne.0) then
+                        call error_mesg( 'socrates_init', &
```

Comment:
No - the code will run fine. The reason for it being a warning is that, if you imagine your radiation scheme is called every 8 hours, then your coszen (time-averaged over 8 hours each time) will peak at 4 different longitude locations each day (obviously the daily-mean will be zonally-symmetric, but each time the radiation is called, there will be a longitudinal peak). Because 8 hours neatly fits into 24, then these 4 locations will be the same each day. I was quite keen on this, as I liked the fact that each day was repeatable. However, it has been argues that you should, in fact, use a `dt_rad` that doesn't fit equally into one day, so that those `n` longitudinal peaks per day move in longitude from one day to the next. I put the warning in there to alert the user to the fact that they would be doing this. But in answer to your question, the code will still run fine and produce sensible answers etc if you ignore this warning. We could, perhaps, consider making it a note instead of a warning, so that it looks less sinister?

Reply:
I think it's ok as it is then.  The reason it came up is the use case we discussed this morning:  I want zonally averaged insolation, but with a seasonal cycle.  i.e. eliminate the diurnal cycle while maintaining the other properties of the astrophys.  
This works fine when using a calendar, but not when you want to change the length of the year of the planet.  To achieve (I hope!) zonally average insolation for arbitrary orbital period (year length) and rotation rate (~= 2π/day length for earth, but not true for all planets) I created this helper function in my python script:

```python
SECONDS_PER_DAY=86400.0
def setup_calendar(exp, orbital_period, omega):
    """Create an idealised calendar: 
    * NH equinox at 75% of year
    * N days long
    * rotation rate omega scaled b
    * average insolation over a day period
    * set the length of the run to be 1 orbital period
    """
    orbital_period = orbital_period*SECONDS_PER_DAY
    orbital_rate = 2*np.pi/orbital_period
    seconds_per_sol = abs(2*np.pi / (orbital_rate - omega))
    exp.update_namelist({
        'main_nml': {
            'calendar' : 'no_calendar',
            'days': int(orbital_period/SECONDS_PER_DAY)
        },
        'socrates_rad_nml': {
            'dt_rad_avg': seconds_per_sol,
            'equinox_day': 0.75,
        },
        'constants_nml': {
            'omega': omega,
            'orbital_period': orbital_period
        }
    })
```

As you call `length_of_day()` in this check, it may not even be an integer number of seconds for an exoplanet.